service.subtitles.napisy24pl/service.py

```diff
@@ -0,0 +1,162 @@
+    item['rar'] = False
+    item['year'] = xbmc.getInfoLabel("VideoPlayer.Year")  # Year
+    item['season'] = str(xbmc.getInfoLabel("VideoPlayer.Season"))  # Season
+    item['episode'] = str(xbmc.getInfoLabel("VideoPlayer.Episode"))  # Episode
+    item['tvshow'] = normalizeString(xbmc.getInfoLabel("VideoPlayer.TVshowtitle"))  # Show
+    item['title'] = normalizeString(xbmc.getInfoLabel("VideoPlayer.OriginalTitle"))  # try to get original title
+    item['file_original_path'] = unquote(xbmc.Player().getPlayingFile())  # Full path of a playing file
+    item['file_original_name'] = os.path.basename(item['file_original_path'])  # Name of playing file
+    item['3let_language'] = []
+    item['preferredlanguage'] = unquote(params.get('preferredlanguage', ''))
+    item['preferredlanguage'] = xbmc.convertLanguage(item['preferredlanguage'], xbmc.ISO_639_2)
+
+    for lang in unquote(params['languages']).split(","):
+        item['3let_language'].append(xbmc.convertLanguage(lang, xbmc.ISO_639_2))
+
+    if item['title'] == "":
+        log("VideoPlayer.OriginalTitle not found")
+        item['title'] = normalizeString(xbmc.getInfoLabel("VideoPlayer.Title"))  # no original title, get just Title
+
+    if params['action'] == 'manualsearch':
+        if item['season'] != '' or item['episode']:
+            item['tvshow'] = unquote(params['searchstring'])
+        else:
+            item['title'] = unquote(params['searchstring'])
+
+    for lang in unquote(params['languages']).split(","):
+        item['3let_language'].append(xbmc.convertLanguage(lang, xbmc.ISO_639_2))
+
+    log("Item before cleaning: \n    %s" % item)
+
+    # clean title + tvshow params
+    clean_title(item)
+    parse_rls_title(item)
+
+    if item['episode'].lower().find("s") > -1:  # Check if season is "Special"
+        item['season'] = "0"
+        item['episode'] = item['episode'][-1:]
+
+    if (item['file_original_path'].find("http") > -1):
+        item['temp'] = True
+
+    elif (item['file_original_path'].find("rar://") > -1):
+        item['rar'] = True
+        item['file_original_path'] = os.path.dirname(item['file_original_path'][6:])
+
+    elif (item['file_original_path'].find("stack://") > -1):
+        stackPath = item['file_original_path'].split(" , ")
+        item['file_original_path'] = stackPath[0][8:]
+
+    item["file_original_size"] = xbmcvfs.File(item["file_original_path"]).size()
+
+    log("item: %s" % (item))
+
+    search(item)
+
+elif params['action'] == 'download':
+    ## we pickup all our arguments sent from def search()
+    subs = download(params["id"])
+
+    ## we can return more than one subtitle for multi CD versions, for now we are still working out how to handle that in XBMC core
+    for sub in subs:
+        listitem = xbmcgui.ListItem(label=sub)
+        xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=sub, listitem=listitem, isFolder=False)
+
+xbmcplugin.endOfDirectory(int(sys.argv[1]))  ## send end of directory to XBMC
```

Comment:
Usually it is a good practice to keep the entrypoint files as small as possible. Normally you just import your main code and run:

```
from resources lib import my_service

my_service.run()
```

Reply:
I don't see any difference in this suggestion.

Reply:
Entrypoints are the only files that are not compiled to bytecode by kodi's python interpreter. If you keep the entrypoint as small as possible and move the main codebase to a module startup times will be a bit faster. This change is not mandatory.